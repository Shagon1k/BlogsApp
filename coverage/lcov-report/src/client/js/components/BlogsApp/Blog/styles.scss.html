<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/client/js/components/BlogsApp/Blog/styles.scss</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\..\..\..\index.html">All files</a> / <a href="index.html">src/client/js/components/BlogsApp/Blog</a> styles.scss
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/0</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">.blogBox {
	position: relative;
	box-sizing: border-box;
	width: 100%;
	background: #ccd2dd;
	border-radius: 5px;
	padding: 18px;
	margin-bottom: 10px;
&nbsp;
	.blogInfo {
		display: flex;
		flex-flow: row wrap;
		align-items: flex-start;
&nbsp;
		dt {
			flex-basis: 20%;
			margin-bottom: 5px;
			font-weight: bold;
		}
&nbsp;
		dd {
			flex-basis: 80%;
			margin-bottom: 10px;
			text-align: justify;
		}
	}
	
	.removeBlog {
		position: absolute;
		top: 12px;
		right: 12px;
		font-size: 20px;
		font-weight: bold;
		color: #961e1e;
	}
&nbsp;
	.blogDetailsLink {
		position: absolute;
		right: 12px;
		bottom: 12px;
		font-weight: bold;
		color: #961e1e;
	}
}</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Mar 03 2018 15:03:49 GMT+0300 (Belarus Standard Time)
</div>
</div>
<script src="..\..\..\..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\..\..\..\sorter.js"></script>
</body>
</html>
