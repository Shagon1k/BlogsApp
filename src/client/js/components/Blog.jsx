import React from "react";
import Portal from "./Portal.jsx";
import DeleteBlogModal from "./modals/DeleteBlogModal.jsx"

class Blog extends React.Component {
	constructor(props) {
		super(props);

		this.state = {
			showDeleteBlogModal: false
		}
	}

	onDelete() {
		this.props.onDelete(this.props.blog);
	}

	closeDeleteBlogModal() {
		this.setState({ showDeleteBlogModal: false })
	}

	openDeleteBlogModal(e) {
		e.preventDefault();
		this.setState({ showDeleteBlogModal: true })
	}

	render() {
		return (
			<li className="blogBox">
				<dl className="blogInfo">
					<dt> Title: </dt>
					<dd> {this.props.blog.title} </dd>
					<dt> Author: </dt>
					<dd> {this.props.blog.author} </dd>
					<dt> Message: </dt>
					<dd> {this.props.blog.message}</dd>
				</dl>
				<a href="#" className="removeBlog" onClick={(e) => this.openDeleteBlogModal(e)}> X </a>
				{ this.state.showDeleteBlogModal &&
					<Portal>
						<DeleteBlogModal 
							blog= {this.props.blog} 
							closeDeleteBlogModal={() => this.closeDeleteBlogModal()} 
							acceptDelete={() => this.onDelete()}
						/>
					</Portal>
				}
			</li>
			);
	}
}

export default Blog;