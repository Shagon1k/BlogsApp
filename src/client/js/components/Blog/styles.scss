.blogBox {
	position: relative;
	box-sizing: border-box;
	width: 100%;
	background: #ccd2dd;
	border-radius: 5px;
	padding: 18px;
	margin-bottom: 10px;

	.blogInfo {
		display: flex;
		flex-flow: row wrap;
		align-items: flex-start;

		dt {
			flex-basis: 20%;
			margin-bottom: 5px;
			font-weight: bold;
		}

		dd {
			flex-basis: 80%;
			margin-bottom: 10px;
			text-align: justify;
		}
	}
	
	.removeBlog {
		position: absolute;
		top: 12px;
		right: 12px;
		font-size: 20px;
		font-weight: bold;
		color: #961e1e;
	}

	.blogDetailsLink {
		position: absolute;
		right: 12px;
		bottom: 12px;
		font-weight: bold;
		color: #961e1e;
	}
}