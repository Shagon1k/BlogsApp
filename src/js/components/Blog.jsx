import React from "react";

class Blog extends React.Component {
	constructor(props) {
		super(props);

		this.onDelete = this.onDelete.bind(this);
	}

	onDelete(e) {
		e.preventDefault();
		this.props.onDelete(this.props.blog);
	}

	render() {
		return (
			<li className="blogBox">
				<dl className="blogInfo">
					<dt> Title: </dt>
					<dd> {this.props.blog.title} </dd>
					<dt> Author: </dt>
					<dd> {this.props.blog.author} </dd>
					<dt> Message: </dt>
					<dd> {this.props.blog.message}</dd>
				</dl>
				<a href="#" className="removeBlog" onClick={ this.onDelete }> X </a>
			</li>
			);
	}
}

export default Blog;