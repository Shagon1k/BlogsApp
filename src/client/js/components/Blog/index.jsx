import React from 'react';
import './styles.scss';

const Blog = ({blog, onDelete}) => (
	<li className="blogBox">
		<dl className="blogInfo">
			<dt> Title: </dt>
			<dd> {blog.title} </dd>
			<dt> Author: </dt>
			<dd> {blog.author} </dd>
			<dt> Message: </dt>
			<dd> {blog.message}</dd>
		</dl>
		<a href="#" className="removeBlog" onClick={() => onDelete()}> X </a>
	</li>
)

export default Blog;