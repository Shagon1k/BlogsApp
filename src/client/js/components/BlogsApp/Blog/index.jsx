import React from 'react';
import { Link } from 'react-router-dom';
import './styles.scss';

const Blog = ({blog, blogId}) => (
	<li className="blogBox">
		<dl className="blogInfo">
			<dt> Title: </dt>
			<dd> {blog.title} </dd>
			<dt> Author: </dt>
			<dd> {blog.author} </dd>
			<dt> Message: </dt>
			<dd> {blog.message}</dd>
		</dl>
		<Link to={`/blogs/remove/${blogId}`} className="removeBlog"> X </Link>
		<Link className="blogDetailsLink" to={`/blog/${blogId}`}> Blog details </Link>
	</li>
)

export default Blog;