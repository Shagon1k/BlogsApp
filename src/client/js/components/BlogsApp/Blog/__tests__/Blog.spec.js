import React from 'react';
import Blog from '../index.jsx';

describe('Blog component', () => {
	const dummyBlog = {
		title: 'Ololo',
		author: 'Ololosh',
		message: 'Olomessage'
	};
	const dummyBlogId = 123;

	it('renders correctly', () => {
		const wrapper = shallow(<Blog blog={dummyBlog} id={dummyBlogId}/>);
		expect(wrapper).toMatchSnapshot();
	});
})