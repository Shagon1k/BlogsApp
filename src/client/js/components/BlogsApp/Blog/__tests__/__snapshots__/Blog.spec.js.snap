// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog component renders correctly 1`] = `
<li
  className="blogBox"
>
  <dl
    className="blogInfo"
  >
    <dt>
       Title: 
    </dt>
    <dd>
       
      Ololo
       
    </dd>
    <dt>
       Author: 
    </dt>
    <dd>
       
      Ololosh
       
    </dd>
    <dt>
       Message: 
    </dt>
    <dd>
       
      Olomessage
    </dd>
  </dl>
  <Link
    className="removeBlog"
    replace={false}
    to="/blogs/remove/undefined"
  >
     X 
  </Link>
  <Link
    className="blogDetailsLink"
    replace={false}
    to="/blog/undefined"
  >
     Blog details 
  </Link>
</li>
`;
